File: src/ffi/window.ts
import { widget_init } from "./common.ts";

export default {
  window_create: widget_init,
  window_end: {
    parameters: ["pointer"],
    result: "void",
  },
  window_show: {
    parameters: ["pointer"],
    result: "void",
  },
} as const;

File: lib/ffi/src/window.rs
use crate::export;
pub use fltk_sys::window::*;

export!(fn window_create(x: i32, y: i32, width: i32, height: i32, title: *const i8) -> *mut Fl_Window, Fl_Window_new);
export!(fn window_end(win: *mut Fl_Window), Fl_Window_end);
export!(fn window_show(win: *mut Fl_Window), Fl_Window_show);
